wagtail/contrib/modeladmin/views.py

```diff
@@ -704,6 +739,20 @@ def dispatch(self, request, *args, **kwargs):
             return redirect(
                 self.url_helper.get_action_url('edit', self.pk_quoted)
             )
+
+        if self.locale:
```

Comment:
should be 

```suggestion
        if getattr(settings, 'WAGTAIL_I18N_ENABLED', False) and issubclass(self.model, TranslatableMixin):
```

since this is the pattern used everywhere else